src/omnisharp/launcher.ts

```diff
@@ -161,21 +161,15 @@ export interface LaunchResult {
 
 export function launchOmniSharp(details: LaunchDetails): Promise<LaunchResult> {
     return new Promise<LaunchResult>((resolve, reject) => {
-        try {
-            return launch(details).then(result => {
-                // async error - when target not not ENEOT
-                result.process.on('error', reject);
-
-                // success after a short freeing event loop
-                setTimeout(function () {
-                    resolve(result);
-                }, 0);
-            }, err => {
-                reject(err);
-            });
-        } catch (err) {
-            reject(err);
-        }
+        launch(details).then(result => {
+            // async error - when target not not ENEOT
+            result.process.on('error', reject);
+
+            // success after a short freeing event loop
+            setTimeout(function () {
+                resolve(result);
+            }, 0);
+        });
```

Comment:
If there is no reject callback AND the promise is not returned anywhere, I'm quite sure that the error will be silent.

This should be the correct way to bubble the error to the caller:
```
launch(details).then(result => {
    // async error - when target not not ENEOT
    result.process.on('error', reject);

    // success after a short freeing event loop
    setTimeout(function () {
        resolve(result);
    }, 0);
}).catch(reject);
```